examples/naive_bayes_dogs.rs

```diff
@@ -0,0 +1,148 @@
+        let mut red_dog_speed = Normal::new(0., 1.);
+
+        let mut white_dog_friendliness = Normal::new(1., 1.);
+        let mut white_dog_furriness = Normal::new(1., 1.);
+        let mut white_dog_speed = Normal::new(-1., 1.);
+
+        // Flip a coin to decide whether to generate a red or white dog.
+        let coin: f64 = rng.gen();
+        let color = if coin < 0.5 { Color::Red } else { Color::White };
+
+        match color {
+            Color::Red => {
+                Dog {
+                    color: Color::Red,
+                    // sample from our normal distributions for each trait
+                    friendliness: red_dog_friendliness.sample(rng),
+                    furriness: red_dog_furriness.sample(rng),
+                    speed: red_dog_speed.sample(rng),
+                }
+            },
+            Color::White => {
+                Dog {
+                    color: Color::White,
+                    friendliness: white_dog_friendliness.sample(rng),
+                    furriness: white_dog_furriness.sample(rng),
+                    speed: white_dog_speed.sample(rng),
+                }
+            },
+        }
+    }
+}
+
+fn main() {
+    let mut randomness = rand::StdRng::new()
+        .expect("we should be able to get an RNG");
+    let rng = &mut randomness;
+
+    let training_set_size = 1000;
+    let test_set_size = 1000;
+
+    // We'll train the model on these dogs
+    let training_dogs = (0..training_set_size)
+        .map(|_| { Dog::rand(rng) })
+        .collect::<Vec<_>>();
+
+    // ... and then use the model to make predictions about these dogs' color
+    // given only their trait measurements.
+    let test_dogs = (0..test_set_size)
+        .map(|_| { Dog::rand(rng) })
+        .collect::<Vec<_>>();
+
+    // The model's `.train` method will take two matrices, each with a row for
+    // each dog in the training set: the rows in the first matrix contain the
+    // trait measurements; the rows in the second are either [1, 0] or [0, 1]
+    // to indicate color.
+    let mut training_matrix = Matrix::new(0, 3, Vec::new());
+    let mut target_matrix = Matrix::new(0, 2, Vec::new());
+
+    // For every dog in the training set,
+    for training_dog in &training_dogs {
+        // add a row to the input matrix for the trait measurements
+        let dog_row = Matrix::new(
+            1, 3,
+            vec![training_dog.friendliness,
+                 training_dog.furriness,
+                 training_dog.speed]
+        );
+        training_matrix = training_matrix.vcat(&dog_row);
+        // add a row to the target matrix indicating color
+        let color_row = match training_dog.color {
+            Color::Red => Matrix::new(1, 2, vec![1., 0.]),
+            Color::White => Matrix::new(1, 2, vec![0., 1.]),
+        };
+        target_matrix = target_matrix.vcat(&color_row);
+    }
+
+    // Train!
+    let mut model = NaiveBayes::<naive_bayes::Gaussian>::new();
```

Comment:
I think we should give some more information to the reader here (briefly of course). Why do we choose `Gaussian`?